### PR TITLE
Add logger for info/warning/error messages handling

### PR DESCRIPTION
**Pull request by olasaadi99**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,8 +10,8 @@ $ ./bin/vpcanalyzer -h
 Usage of vpc-network-config-analyzer:
   -analysis-type string
         Supported analysis types:
-        * all_endpoints  - supported with: txt, md, json, drawio, arch_drawio, debug
-        * all_subnets  - supported with: txt, md, json, drawio, arch_drawio
+        * all_endpoints  - supported with: txt, md, json, drawio, arch_drawio, svg, arch_svg, html, arch_html, debug
+        * all_subnets  - supported with: txt, md, json, drawio, arch_drawio, svg, arch_svg, html, arch_html
```
Comment (adisos) [inline README.md:14]: why is this change related to this PR?

File: pkg/logging/logger.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+// Warnf writes a warning message to the log (unless DefaultLogger verbosity is set to LowVerbosity)
+func Warnf(format string, o ...interface{}) {
+	if Logger.verbosity >= MediumVerbosity {
+		pc, _, _, _ := runtime.Caller(1)
+		details := runtime.FuncForPC(pc)
+		Logger.l.Printf("WARN	%s	%s", details.Name(), fmt.Sprintf(format, o...))
+	}
+}
```
Comment (adisos) [inline pkg/logging/logger.go:81]: can you show examples for logger output with `details` for these functions?

File: README.md (hunk 1 above)
Comment (olasaadi99) [reply README.md:14]: not related, it was not added to README by the one who added these lines to parse_args

Comment (adisos) [reply README.md:14]: oh. ok

File: pkg/logging/logger.go (hunk 2 above)
Comment (adisos) [reply pkg/logging/logger.go:81]: I think it will be more readable without the details.. simply have:
```
WARN  ignoring floatingIP public-gw-ky: target.ResourceType public_gateway is not supported (only networkInterfaceResourceType supported)
```

File: pkg/logging/logger.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+type DefaultLogger struct {
+	verbosity Verbosity
+	l         *log.Logger
+}
+
+// NewDefaultLogger creates an instance of DefaultLogger with the highest verbosity.
+func NewDefaultLogger() *DefaultLogger {
+	return NewDefaultLoggerWithVerbosity(HighVerbosity)
+}
+
```
Comment (zivnevo) [inline pkg/logging/logger.go:38]: Doesn't seem to be used

File: pkg/logging/logger.go
Comment (zivnevo) [inline pkg/logging/logger.go:1]: Please review all exported symbols and decide if they need to be exported

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/logging/logger.go (hunk 2 above)
Comment (olasaadi99) [reply pkg/logging/logger.go:81]: what about returning error and debug , should I verbose these details?

Comment (zivnevo) [reply pkg/logging/logger.go]: I would leave details only for debug

Comment (olasaadi99) [reply pkg/logging/logger.go]: I think I would also leave details in return error (we still do not use it, will do next pr) it will print the error just when in high verbosity mode, used for debugging also

**Review by zivnevo — APPROVED** [review]
LGTM

File: pkg/logging/logger.go (hunk 2 above)
Comment (adisos) [reply pkg/logging/logger.go]: I'd prefer to keep the returned errors as is, not through logger.. 
It will be difficult to maintain consistency across all project files, and through new development, to remember returning the error from this logger function...
With a simple grep one can find the error message, so no need to provide the exact function name.

**Review by adisos — APPROVED** [review]
